MoreLinq.Test/EndsWithTest.cs

```diff
@@ -91,5 +91,15 @@ public void EndsWithUsesSpecifiedEqualityComparerOrDefault()
             Assert.False(first.EndsWith(second, EqualityComparer.Create<int>(delegate { return false; })));
             Assert.True(first.EndsWith(second, EqualityComparer.Create<int>(delegate { return true; })));
         }
+
+#if IREADONLY
+        [TestCase(new[] { 1, 2, 3 }, new[] { 2, 3 }, ExpectedResult = true)]
+        [TestCase(new[] { 1, 2, 3 }, new[] { 1, 2, 3 }, ExpectedResult = true)]
+        [TestCase(new[] { 1, 2, 3 }, new[] { 0, 1, 2, 3 }, ExpectedResult = false)]
+        public bool EndsWithWithReadOnlyIntegers(IReadOnlyList<int> first, IReadOnlyList<int> second)
```

Comment:
Is this test asserting anything? If I comment out the optimisation for `IReadOnlyCollection<>`, it still passes and which doesn't seem right.

Reply:
It makes sure that the optimization for `IReadOnlyCollection<>` does not cause it to fail. Is that worth something?

Reply:
Yes but the first would be that it is optimised for `IReadOnlyCollection<>` in the first place. How do we know that's covered. As I [demonstrated](https://github.com/morelinq/MoreLINQ/pull/372#discussion_r165290418), someone could come along and comment out that bit or refactor the optimisation out accidentally and tests would let that slip.